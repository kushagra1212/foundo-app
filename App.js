import 'react-native-gesture-handler';
import { StatusBar } from 'expo-status-bar';
import { Link, NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import Toast from 'react-native-toast-message';
import AuthScreen from './src/screens/authScreens/AuthScreen';
import HomeScreen from './src/screens/HomeScreen';
import { toastConfig } from './src/configs/toastConfig';
import { useFonts } from 'expo-font';
import { ROBOTO_FONTS } from './src/assets/fonts';
import { store } from './src/redux/store';
import { Provider, useDispatch, useSelector } from 'react-redux';
import { userLoggedIn } from './src/redux/services/auth-service';
import { useEffect, useState } from 'react';
import { routesConfig } from './src/configs/routesConfig';
import {
  selectCurrentUser,
  setCredentials,
} from './src/redux/slices/authSlice';
import { StyleSheet, Text, View, ActivityIndicator } from 'react-native';
import { BASE_URL } from '@env';
const Stack = createNativeStackNavigator();
import * as Linking from 'expo-linking';
import { SafeAreaView } from 'react-native-safe-area-context';
import { COLORS } from './src/constants/theme';
import { GestureHandlerRootView } from 'react-native-gesture-handler';
import { updateAddItemDetailsScreenStatus } from './src/redux/slices/sreenSilce';
const Foundo = () => {
  const [isfontLoaded] = useFonts(ROBOTO_FONTS);
  const [appLoaded, setAppLoaded] = useState(false);
  const [isLoggedIn, setIsloggedIn] = useState(false);
  const [isLoading, setIsLoading] = useState(true);
  const dispatch = useDispatch();
  const url = Linking.useURL();
  useEffect(() => {
    let flag = true;

    userLoggedIn()
      .then((res) => {
        if (!flag) return;
        console.log(res);
        if (res.isLoggedIn === true) {
          dispatch(
            setCredentials({
              user: res.user,
              jwtToken: res.token,
            })
          );
          setIsloggedIn(res.isLoggedIn);
        } else if (url) {
          const { path } = Linking.parse(url);
          const pathArray = path?.split('/');
          if (pathArray !== undefined && pathArray?.length >= 2) {
            dispatch(
              setCredentials({
                user: { email: pathArray[pathArray?.length - 2] },
                jwtResetToken: pathArray[pathArray?.length - 1],
                jwtToken: '',
              })
            );
          }
        }
        setIsLoading(false);
      })
      .catch((err) => console.log(err));

    setAppLoaded(true);
    return () => (flag = false);
  }, [url]);
  if (!isfontLoaded || !appLoaded) {
    return null;
  }

  const prefix = Linking.createURL('app');
  if (isLoading) {
    return (
      <SafeAreaView mode="margin">
        <ActivityIndicator
          style={{ marginTop: '50%', transform: [{ scale: 3 }] }}
          size="large"
          color={COLORS.redPrimary}
        />
      </SafeAreaView>
    );
  }
  return (
    <NavigationContainer linking={{ prefixes: [prefix], config: routesConfig }}>
      <Stack.Navigator initialRouteName={'Home'}>
        <Stack.Screen
          options={{ headerShown: false }}
          name="Auth"
          component={AuthScreen}
        />
        <Stack.Screen
          options={{ headerShown: false }}
          name="Home"
          component={HomeScreen}
        />
      </Stack.Navigator>

      <Toast config={toastConfig} visibilityTime={2000} />
    </NavigationContainer>
  );
};
export default function App() {
  return (
    <Provider store={store}>
      <Foundo />
    </Provider>
  );
}

const styles = StyleSheet.create({
  container: {
    height: '100%',
    width: '100%',
  },
});
