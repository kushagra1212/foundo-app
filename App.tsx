import { ApiProvider } from '@reduxjs/toolkit/dist/query/react';
import * as Notifications from 'expo-notifications';
import * as SplashScreen from 'expo-splash-screen';
import { StatusBar } from 'expo-status-bar';
import { useEffect } from 'react';
import { ActivityIndicator } from 'react-native';
import { SafeAreaView } from 'react-native-safe-area-context';
import { Provider } from 'react-redux';

import Error from './src/components/Error';
import Foundo from './src/components/Foundo';
import { LoadFoundo } from './src/components/molecules/Hooks/Auth/LoadFoundo';
import { COLORS } from './src/constants/theme';
import { api } from './src/redux/services/api-service';
import { store } from './src/redux/store';
// Re-export the default UI

SplashScreen.preventAutoHideAsync();
// const errorHandler = (error: Error, stackTrace: string) => {
//   /* Log the error to an error reporting service */
//   // console.log(error, stackTrace);
// };
Notifications.setNotificationHandler({
  handleNotification: async () => ({
    shouldShowAlert: true,
    shouldPlaySound: true,
    shouldSetBadge: false,
  }),
});

const App = () => {
  const [isLoaded, error, credentials] = LoadFoundo();
  useEffect(() => {
    if (isLoaded) {
      (async () => {
        await SplashScreen.hideAsync();
      })();
    }
  }, [isLoaded]);

  if (!isLoaded) {
    return (
      <SafeAreaView>
        <StatusBar style="dark" />
        <ActivityIndicator size="large" color={COLORS.redPrimary} />
      </SafeAreaView>
    );
  }

  if (error) {
    return <Error error={error} resetError={() => {}} />;
  }

  return (
    <ApiProvider api={api}>
      <Provider store={store}>
        <StatusBar style="dark" />
        <Foundo credentials={credentials} />
      </Provider>
    </ApiProvider>
  );
};
export default App;
