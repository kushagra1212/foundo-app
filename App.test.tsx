import renderer from 'react-test-renderer';

import App from './App';

describe('<App />', () => {
  it('should renders correctly', async () => {
    await renderer.act(async () => {
      renderer.create(<App />);
    });
  });
});
