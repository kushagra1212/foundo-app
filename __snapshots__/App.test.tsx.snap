// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should renders correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#fafafa",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      {
        "marginHorizontal": 16,
      }
    }
  >
    <Text
      style={
        {
          "color": "#000",
          "fontSize": 48,
          "fontWeight": "300",
          "paddingBottom": 16,
        }
      }
    >
      Oops!
    </Text>
    <Text
      style={
        {
          "color": "#000",
          "fontSize": 32,
          "fontWeight": "800",
        }
      }
    >
      There's an error
    </Text>
    <Text
      style={
        {
          "paddingVertical": 16,
        }
      }
    >
      Error: expo-linking needs access to the expo-constants manifest (app.json or app.config.js) to determine what URI scheme to use. Setup the manifest and rebuild: https://github.com/expo/expo/blob/main/packages/expo-constants/README.md
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#2196f3",
          "borderRadius": 50,
          "opacity": 1,
          "padding": 16,
        }
      }
    >
      <Text
        style={
          {
            "color": "#fff",
            "fontWeight": "600",
            "textAlign": "center",
          }
        }
      >
        Try again
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
